prefs.js

```diff
@@ -0,0 +1,126 @@
+// Based on https://github.com/ubuntu/gnome-shell-extension-appindicator
+
+"use strict";
+
+const {
+    Gio,
+    Gtk,
+    GObject
+} = imports.gi;
+
+const ExtensionUtils = imports.misc.extensionUtils;
+const Me = ExtensionUtils.getCurrentExtension();
+
+
+const WMCLASS_LIST = 'by-class';
+const BLACKLIST_ENABLED = 'enable-blacklist';
+
+function init() {}
+
+function buildPrefsWidget() {
+    this.settings = ExtensionUtils.getSettings(
+        "org.gnome.shell.extensions.noannoyance"
+    );
+
+    let settingsBox = new Gtk.Box({
+        orientation: Gtk.Orientation.VERTICAL,
+        spacing: 8,
+        margin_bottom: 60,
+    });
+
+    let enableBlacklistBox = new Gtk.Box({
+        orientation: Gtk.Orientation.HORIZONTAL,
+    });
+    let wmClassBox = new Gtk.Box({
+        orientation: Gtk.Orientation.HORIZONTAL
+    });
+
+    let toggleLabel = new Gtk.Label({
+        label: "Enable Blacklist",
+        halign: Gtk.Align.START,
+        hexpand: true,
+        visible: true,
+    });
+
+    let toggle = new Gtk.Switch({
+        active: this.settings.get_boolean(BLACKLIST_ENABLED),
+        halign: Gtk.Align.END,
+        visible: true,
+    });
+
+    enableBlacklistBox.append(toggleLabel);
+    enableBlacklistBox.append(toggle);
+
+    this.settings.bind(
+        BLACKLIST_ENABLED,
+        toggle,
+        "active",
+        Gio.SettingsBindFlags.DEFAULT
+    );
+
+    settingsBox.append(enableBlacklistBox);
+
+    const customListStore = new Gtk.ListStore();
+    customListStore.set_column_types([GObject.TYPE_STRING]);
+    const customInitArray = settings.get_strv(WMCLASS_LIST);
+    for (let i = 0; i < customInitArray.length; i++) {
+        customListStore.set(customListStore.append(), [0], [customInitArray[i]]);
+    }
+    customListStore.append();
+
+    const customTreeView = new Gtk.TreeView({
+        model: customListStore,
+        hexpand: true,
+        vexpand: true,
+    });
+
+    const indicatorIdColumn = new Gtk.TreeViewColumn({
+        title: "WM__CLASS List",
```

Comment:
Maybe add a hint to the string on how to get the wmclass?

What works for both wayland and x11 windows is >Alt + F2 -> lg -> Windows.